pineapple-dependency-loader/src/main/java/sh/miles/pineapple/dependencyloader/PaperDependencyLoader.java

```diff
@@ -0,0 +1,48 @@
+package sh.miles.pineapple.dependencyloader;
+
+import com.google.gson.Gson;
+import io.papermc.paper.plugin.loader.PluginClasspathBuilder;
+import io.papermc.paper.plugin.loader.PluginLoader;
+import io.papermc.paper.plugin.loader.library.impl.MavenLibraryResolver;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public class PaperDependencyLoader implements PluginLoader {
```

Comment:
This class needs to be documented and this class should also probably have a different name. I much prefer a name including Pineapple or being more descriptive e.g. JsonDependencyLoader or FileDependencyLoader if you want to go more generic.